src/org/exist/xquery/FunctionCall.java

```diff
@@ -362,15 +362,20 @@ public void accept(ExpressionVisitor visitor) {
         visitor.visitFunctionCall(this);
     }
 
-    private class DeferredFunctionCallImpl extends DeferredFunctionCall {
+    private static class DeferredFunctionCallImpl extends DeferredFunctionCall {
```

Comment:
My preference would like be for (1) as that means less change and that was working before.